### PR TITLE
Improve documentation to bootstrap straight package manager

### LINKED ISSUE #315: [craftedv2beta] Improve documentation for using straight.el
I finally managed to setup `craftedv2beta` to use `straight` instead of the builtin package manager. The doc alone _(info "(crafted-emacs)Using alternate package managers"')_ was not enough to get started. Here is my setup:

early-init.el:

```emacs-lisp
(setq crafted-emacs-home "~/crafted-emacs-v2"
      crafted-bootstrap-directory (expand-file-name "bootstrap" crafted-emacs-home)
      crafted-config-path user-emacs-directory)
(load (expand-file-name "crafted-package" crafted-bootstrap-directory))
(setq crafted-package-system 'straight)
(crafted-package-bootstrap)

;; Adds crafted-emacs modules to the `load-path', sets up a module
;; writing template, sets the `crafted-emacs-home' variable, unless not already set.
;; We set it in `early-init' explicitly.
(load (expand-file-name "modules/crafted-init-config" crafted-emacs-home))

;; Now we can require this and configure `straight' as explained in `(info "(crafted-emacs)Using alternate package managers"'
(require 'crafted-package-config)

(setq crafted-package-installer #'straight-use-package)
(setq crafted-package-installed-predicate #'straight--installed-p)

;; Should we have an `crafted-early-init-straight-config' containing the code in this file ?
;; (load "~/crafted-emacs-v2/modules/crafted-early-init-config")
```

init.el:

```emacs-lisp
;;; Initial phase.

;; Load the custom file if it exists.  Among other settings, this will
;; have the list `package-selected-packages', so we need to load that
;; before adding more packages.  The value of the `custom-file'
;; variable must be set appropriately, by default the value is nil.
;; This can be done here, or in the early-init.el file.
(setq custom-file (expand-file-name "custom.el" user-emacs-directory))
(when (and custom-file
           (file-exists-p custom-file))
  (load custom-file nil 'nomessage))

;; Adds crafted-emacs modules to the `load-path', sets up a module
;; writing template, sets the `crafted-emacs-home' variable.
;; (load (expand-file-name "modules/crafted-init-config" crafted-emacs-home))
;; already done in early-init.el, to configure package-system.

;;; Packages phase

;; Collect list of packages to install.
(require 'crafted-completion-packages)

;; Install the packages listed in the `package-selected-packages' list. CHANGE: Must use
;; `crafted-package-install-selected-packages' instead of
;; `package-install-selected-packages'
(crafted-package-install-selected-packages)

;; Rest same as in example init.el
```

Is this the intended way to use _straight_ as package manager? If so, i would recommend to provide these two files as additional *init.el examples in the _examples_ directory and refer to them in the docs.

### PR DESCRIPTION
**Pull request by monora**
fixes #315

### CODE REVIEW

**Comment by jeffbowman** [comment]
The info file needs to be regenerated.  

The `crafted-init-config` module sets the `crafted-emacs-home` variable, it does so by deducing the path where the `crafted-emacs` project code lives, as long as the value hasn't already been set.  So, it is safe to set in the `early-init.el` file, but I wonder if we might want to use a similar method. 

I'm not sold on copying the example to the `custom-modules` path and then loading it from there when it could easily be the user's `early-init.el` file by itself.  Doing so would mean the `crafted-emacs-home` value doesn't need to be set there, so less confusion on when/where/how to set that variable.  Additionally, if the `straight.el` maintainer devises a new method to bootstrap `straight`, then the example becomes a source of bugs, whereas depending on the user to own their own configuration means we don't need to track that process.  Technically, it is just **an example**, but in the face of such a change, someone will file a bug report that the example is wrong.  

On the other side of that coin, having a working example like this would be valuable and helpful to many, and the risk of changes isn't actually that high... sooo... I'm mostly in favor of merging this PR, but perhaps @jvdydev will share some thoughts as well.

**Comment by jvdydev** [comment]
I think the idea of providing an example file for straight.el is generally a good idea (long-term I want to expand the examples section, but that should do for now).
Updating a broken example because of bootstrapping changes feels much less overhead than a module (maybe with a nice reminder that we don't officially support other package managers).

I agree on the copying an example and using it as a module (being equally not sold on it).
I think of the examples as a "copy in as my early-init/init and play around/get started"-type deal compared to the modules which are configuration blocks that are essential to the day-to-day usage.

I think personally, the only change on the docs should be maybe adding how to load the `crafted-package-config` (potentially setting `crafted-emacs-home` in the process, I'm open for either way on that one).
Otherwise the docs section above the "Here is an example using straight.el" is quite verbose on where those variables come from (in my opinion).

File: docs/crafted-emacs.org [hunk 1]
```diff
@@ -131,11 +131,28 @@ for the code examples.
   install packages:
 
   #+begin_src emacs-lisp
```
Comment (jvdydev) [inline docs/crafted-emacs.org:133]: Maybe adding how to load `crafted-package-config` here would be nice

File: docs/crafted-emacs.org [hunk 2]
```diff
@@ -131,11 +131,28 @@ for the code examples.
...
-    (setq crafted-package-installer #'straight-use-package)
-    (setq crafted-package-installed-predicate #'straight--installed-p)
-    (crafted-package-install-package-list)
+  (setq crafted-package-installer #'straight-use-package)
+  (setq crafted-package-installed-predicate #'straight--installed-p)
+  (crafted-package-install-package-list)
   #+end_src
 
+  You can find an example to bootstrap straight in the examples directory:
```
Comment (jvdydev) [inline docs/crafted-emacs.org:139]: I'd personally keep the "You can find an example" part, but remove the copying to custom-modules block.

File: examples/crafted-early-init-straight.el [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+;; Copyright (C) 2023
+;; SPDX-License-Identifier: MIT
+
+;; Author: System Crafters Community
+
+;;; Commentary:
+
+;; Code to bootstrap straight package manager
+;;
+;; Copy it to your custom-modules folder and load it in early-init.el like this:
```
Comment (jvdydev) [inline examples/crafted-early-init-straight.el:12]: Same here on the copying to custom-modules

**Review by jvdydev — COMMENTED** [review]
Basically what my previous comment said, just with local comments, see above for some of the reasoning.

**Comment by monora** [comment]
With 267b5c4 I hope I resolved all comments. I reworded the doc again a little. Please have a look.

File: docs/crafted-emacs.org [hunk 4]
```diff
@@ -144,6 +150,23 @@ for the code examples.
     (straight-use-package 'corfu)
   #+end_src
 
+  You can find a full example to bootstrap straight in the examples directory:
+
+  #+begin_src emacs-lisp
+    (find-file "../examples/crafted-early-init-straight.el")
```
Comment (jeffbowman) [inline docs/crafted-emacs.org:156]:
```suggestion
    [[file:../examples/crafted-early-init-straight.el][crafted-early-init-straight.el]]
```

This should be a link as this document could be viewed online or in info.  (and therefore don't need the src block tags either. )

File: docs/crafted-emacs.org [hunk 5]
```diff
@@ -144,6 +150,23 @@ for the code examples.
...
+  #+end_src
+
+  Assuming you copied it to your ~custom-modules~ folder, you can load it in
```
Comment (jeffbowman) [inline docs/crafted-emacs.org:159]: I might consider this an alternate approach.  The example file could be copied (or merged) with/as the user's `early-init.el` file

**Review by jeffbowman — CHANGES_REQUESTED** [review]
A few more changes.  This is good work and I really appreciate the effort you are putting into it.  Thanks for being willing to work with us on this PR! 

cc: @jvdydev

**Comment by jeffbowman** [comment]
> Do you squash, when you merge?

I usually just rebase, but you are welcome to squash if you wish.

File: docs/crafted-emacs.org [hunk 6]
```diff
@@ -144,6 +150,23 @@ for the code examples.
...
+  your ~early-init~.el like this:
+
+  #+begin_src emacs-lisp
+  (setq crafted-emacs-home "~/crafted-emacs")
+
+  (load (expand-file-name "modules/crafted-package-config" crafted-emacs-home))
```
[the remark this answers is not in the document]
Comment (monora) [reply docs/crafted-emacs.org]: Done with fa8e2aed

**Comment by monora** [comment]
> > Do you squash, when you merge?
> 
> I usually just rebase, but you are welcome to squash if you wish.

OK, than I would squash manually, so that you can merge if you are satisfied with my last changes.

**Review by jeffbowman — APPROVED** [review]

**Comment by monora** [comment]
Squashed to one commit. Ready to merge.

**Pull request merged**